From 24bd3d5b2ea519900712208aabddee6283c20612 Mon Sep 17 00:00:00 2001
From: Friedemann Kleint <Friedemann.Kleint@qt.io>
Date: Mon, 17 Oct 2022 09:54:44 +0200
Subject: [PATCH] Fix build with OpenGLES

Add the GLES headers to the system headers so that the integer
typedefs are seen, which is required after
f92cd6b5ac0dc3f97d9693443e6ac5cf966b87ec,
50d0c3c94ff66c104de269f09c08109eb9d0eb3a.

Fixes: PYSIDE-2084
Pick-to: 6.4 6.2
Change-Id: I6036d042765b959e0f8d7258d5d5ba322351f8ec
---

diff --git a/sources/shiboken6/ApiExtractor/clangparser/clangbuilder.cpp b/sources/shiboken6/ApiExtractor/clangparser/clangbuilder.cpp
index 7887457..512f1a8 100644
--- a/sources/shiboken6/ApiExtractor/clangparser/clangbuilder.cpp
+++ b/sources/shiboken6/ApiExtractor/clangparser/clangbuilder.cpp
@@ -807,6 +807,10 @@
     // Resolve OpenGL typedefs although the header is considered a system header.
     const QString baseName = clang::baseName(fileName);
     if (baseName == u"gl.h"
+        || baseName == u"gl2.h"
+        || baseName == u"gl3.h"
+        || baseName == u"gl31.h"
+        || baseName == u"gl32.h"
         || baseName == u"stdint.h" // Windows: int32_t, uint32_t
         || baseName == u"stddef.h") { // size_t
         return true;
